packages/vue/src/composables/useAuth.ts

```diff
@@ -1,6 +1,10 @@
-import { useActor } from '@xstate/vue';
-import { authService } from '@aws-amplify/ui';
+import { authMachine } from '@aws-amplify/ui';
+import { useActor, useInterpret } from '@xstate/vue';
 
 export const useAuth = () => {
-  return useActor(authService);
+  const service = useInterpret(authMachine, {
+    devTools: process.env.NODE_ENV === 'development',
```

Comment:
Will `process` be defined in the browser?

Reply:
This is normal convention for bundlers since webpack2 to do dead-code elimination by setting `process.env.NODE_ENV = "production"` and evaluating this to `devTools: 'production' === 'development'`, **but I assume this is true for Vite as well**.

Let me check...

Yes, it should work:
https://github.com/vitejs/vite/commit/d7ceabe92efe1c523ee05a6941e3f5042733b2bf

If not, we can remove this.  But I'm still on the fence if this should be a default value in `useAuth` vs. for development we use `<AmplifyProvider services={authService}>` to override it 🤔  We'll need to test out the DX some more – that visualization can be essential to validating flows.

Reply:
@ErikCH Ahhh, I think the issue here is that this is **repeatedly** creating the `authMachine` & returning a service – **so it's not a single machine**.

For React, the `Authenticator` renders once and creates the machine there, passing down the service via context:

https://github.com/aws-amplify/amplify-ui/blob/e9be592f172d6ac2613dd76123d5c5de47357b85/packages/react/src/components/Authenticator/index.tsx#L15-L23

What's the best way to do this with Vue?

Reply:
Should https://github.com/aws-amplify/amplify-ui/blob/main/packages/vue/src/components/authenticator.vue `provide` the service and each sub-screen `inject` it rather than `useAuth`?

Reply:
Provide and inject could work, but it's a little messy trying to provide the service at the authenticator level. I decided instead to use `ref` which creates a reactive variable, that is stored in memory.

Reply:
This should fix Vue from failing to run.